index.js

```diff
@@ -209,16 +209,16 @@ module.exports = (opts = {}) => {
       }
 
       return {
-        AtRule: {
-          'add-mixin': (node, helpers) => {
-            insertMixin(helpers, mixins, node, opts)
-          },
-          'define-mixin': (node, helpers) => {
-            addMixin(helpers, mixins, node)
-            node.remove()
-          },
-          'mixin': (node, helpers) => {
-            insertMixin(helpers, mixins, node, opts)
+        AtRule(node, helpers) {
```

Comment:
But the issue is that `postcss-mixins` is executed *after* `postcss-simple-vars` right now. And the same will keep happening if we subscribe to AtRuleExit I think 🤔

Reply:
> But the issue is that postcss-mixins is executed after postcss-simple-vars right now.

But `postcss-mixins` should be called after `postcss-simple-vars`. Otherwise, other users will not be able to use vars as mixin argument.

You have not a problem with PostCSS 8, but a problem of using the hack to pass argument with `,` to mixin. PostCSS 8 just broke the hack.

Reply:
> Otherwise, other users will not be able to use vars as mixin argument.

I think it will not change anything in this matter.
Let's look at this example:
I assume this order of plugins in config: `[mixins, vars]`
```
@define-mixin a $b, $c { color: $b; margin: $c; }

.selector { $margin: 4px; @mixin a black, $margin; }
```
And because `simple-vars` performs replacements first right now this is what `mixins` plugin 'sees':
```
@define-mixin a $b, $c { color: $b; margin: $c; }

.selector { @mixin a black, 4px; }
```
which is then transformed into:
```
.selector { color: black; margin: 4px; }
```

After my changes the order from config will be preserved so `mixins` plugin will transform the code first ($margin definition would not be present in nodes at this point I think):

```
.selector { $margin: 4px; color: black; margin: $margin; }
```
which will be again transformed into:
```
.selector { color: black; margin: 4px; }
```

So as you can see the typical usage will not change at all. Or maybe you have a specific scenario in mind? If so, could you give some more details?

> You have not a problem with PostCSS 8, but a problem of using the hack to pass argument with , to mixin.

I agree that this is a hack but on the other hand, I also think that there should be some kind of way to pass such parameters to mixins because they are 100% valid CSS values.
What would you say about introducing a new syntax, e.g. `@mixin [a, b], c` to do the same kind of transformation?

Reply:
> I assume this order of plugins in config: [mixins, vars]

There is no order in PostCSS 8. In PostCSS 8 all plugins works together in the same moment by events.

> What would you say about introducing a new syntax, e.g. @mixin [a, b], c to do the same kind of transformation?

We can add `@mixin (a, b), c`

But are we on right direction of simple sources, which is easy to read? Previously, you need to know the plugins order to read styles. Now you need to know tricky syntax.

Reply:
I don't think that adding square brackets would significantly complicate the use of this plugin. However, I would be rather opposed to using parentheses because they might be too closely associated with function calls. In this case, we are definitely closer to a list of values.

Furthermore, I personally perceive this change as a nice addition. Nobody will be forced to use it, but it can be extremely helpful in specific situations.

As for code readability, in my opinion, it mainly depends on the user and the naming of mixins/parameters and brackets would neither improve nor obscure readability.

Mixins themselves are non-standard CSS anyway and it gives a bit more freedom when it comes to introducing new things - without overdoing of course 😉

Reply:
It's a hard question to answer without asking a bigger group of developers 😉

But I really like the function-like approach. I would suggest `asSingleArg(first, second)` though - after all, it would do *exactly* that so it should be self-explanatory.
If we agree on that I'll be happy to rewrite this PR.

Reply:
OK, let’s try `asSingleArg()`